pkg/openstack/loadbalancer.go

```diff
@@ -2156,7 +2156,7 @@ func (lbaas *LbaasV2) UpdateLoadBalancer(ctx context.Context, clusterName string
 
 func compareSecurityGroupRuleAndCreateOpts(rule rules.SecGroupRule, opts rules.CreateOpts) bool {
 	return rule.Direction == string(opts.Direction) &&
-		rule.Protocol == string(opts.Protocol) &&
+		strings.ToLower(rule.Protocol) == strings.ToLower(string(opts.Protocol)) &&
```

Comment:
seems strings.EqualFold can be used instead: https://pkg.go.dev/strings#EqualFold

Reply:
Fixed!